src/DynamicNLPModels.jl

```diff
@@ -144,20 +144,20 @@ function LQDynamicData(
         error("size of Q is not consistent with length of s0")
     end
 
-    if !(sl  <= su)
+    if sum(sl .>= su) > 0
```

Comment:
Let's implement something like this
```julia
function _cmp_arr(op,A,B)
    for i=1:length(A)
        !op(A[i],B[i]) && return false
    end
    return true
end
```

Reply:
See also the function `all`: 
```julia
all(sl .<= su)
```
(but this solution leads to the allocation of one array)